src/main/java/org/rulez/demokracia/PDEngine/VoteRegistry.java

```diff
@@ -28,4 +36,97 @@ public static Vote getByKey(String adminKey) {
 		return votes.get(adminKey);
 	}
 
+    public static void checkVoteName(String voteName) throws ReportedException {
+
+
+		ErrorReason checkValue = checkString(voteName);
+
+		if (checkValue == ErrorReason.STR_SHORT) {
```

Comment:
Just throw the exception in checkString. This way yo won't need these complicated checks and neither the ErrorReason enum, Give the information about the checked field as parameter, like
checkString(voteName, "Vote name")

Also, do not try to assemble the message here. UI is responsible for interactions with user, and will receive something like
["{0} is too short!", "Vote name"]
change ReportedError to have a string for error type and a list of strings for arguments.